src/tests/homework.spec.js

```diff
@@ -0,0 +1,221 @@
+}
+
+async function getPasswordFieldLocator(page) {
+    return page.getByLabel("Heslo");
+}
+async function getControlPasswordFieldLocator(page) {
+    return page.getByLabel("Kontrola hesla");
+}
+async function getRegistraceButtonLocator (page) {
+    return page.getByRole('button', {name: 'Zaregistrovat'});
+}
+
+async function getCurrentUserLocator(page) 
+{
+    const currentUser = page
+        .locator(".navbar-right")
+        .locator("strong");
+    return currentUser;
+}
+async function getLoginInLocator(page) 
+{
+    const loginInLocator = await page.locator(".navbar-right"); 
+    return loginInLocator;
+}
+
+// return texts of error messages
+async function getToastMessageLocator (page) 
+{
+    const errorLocator = await page.locator(".toast-message").textContent();
+    return errorLocator;
+}
+
+async function getFieldErrorLocator(page, thisLabel) 
+{
+    const errorLocator = await page
+        .getByRole("alert")
+        .getByText(thisLabel)
+        .textContent(); 
+    return errorLocator;
+}
+
+//function for navigation to the registration page
+async function goToPage (page) 
+{
+    await page.goto("/registrace");
+    await page.waitForLoadState();
+}
+
+// function for performing user registration
+async function fillRegistrationForm (page, personsData)
+{    
+    const nameField = await getNameFieldLocator(page);
+    const emailField = await getEmailFieldLocator(page);
+    const passwordField = await getPasswordFieldLocator(page);
+    const controlPasswordField = await getControlPasswordFieldLocator(page);
+    const registraceButton = await getRegistraceButtonLocator(page);
+    
+    await nameField.fill(personsData.name);
+    await emailField.fill(personsData.email);
+    await passwordField.fill(personsData.password);
+    await controlPasswordField.fill(personsData.password);
+    
+    await registraceButton.click();
+    console.log("click registration button");  
+}
+
+// return object containing imput fields of the registration form
+async function getformInputFields (page) 
+{
+    const formInputFields = {
+        nameField: await getNameFieldLocator(page),
+        emailField: await getEmailFieldLocator(page),
+        passwordField: await getPasswordFieldLocator(page),
+        controlPasswordField: await getControlPasswordFieldLocator(page),
+    };  
+    return formInputFields;
+}
+
+
+test.describe ("Registration page ", () => {
+
+    test.beforeEach("Go to registration page", async ({ page }) => {
+
+        await goToPage (page);
+        await expect(page).toHaveURL(/registrace/);
+        await expect(page).toHaveTitle("Registrace - Czechitas");
+        console.log("The registration page is open");
+    });
+    
+    
+    test("Verify registration form", async ({ page }) => {
+
+        const formInputFields = await getformInputFields(page);
+        const registraceButton = await getRegistraceButtonLocator(page);
+    
+        for(var handle of Object.values(formInputFields)) {
```

Comment:
Best practice moderního JavaScriptu doporučuje používat const nebo let namísto var. Také chybí mezera za for. Takže by to mohlo vypadat: `for (let handle of Object.values(formInputFields)) {...}`